standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1443,24 +1441,34 @@ public ColumnStatistics getTableStats(final String catName, final String dbName,
         }
         long start = doTrace ? System.nanoTime() : 0;
         Query query = pm.newQuery("javax.jdo.query.SQL", queryText);
-        Object qResult = executeWithArray(query, params, queryText);
-        MetastoreDirectSqlUtils.timingTrace(doTrace, queryText0 + "...)", start, (doTrace ? System.nanoTime() : 0));
-        if (qResult == null) {
-          query.closeAll();
-          return null;
+        try {
```

Comment:
Why are you not using the QueryWrapper here?

Reply:
OK, I will refine this if the `QueryWrapper` seems to be a reasonable choice.

Reply:
Same as below, closing the query earlier may result to exception with `Query result has been closed`.